### PR TITLE
more tests

### PR DESCRIPTION
**Pull request by mkawalek**

### CODE REVIEW

File: server/actions/update-user.js [hunk 1]
```diff
@@ -30,10 +30,8 @@ export default function updateUser({ service, ship, client }: Object, messages:
       if (!_.get(changes, "user['traits_customerio/id']", false)) {
         promises.push(syncAgent.sendAllUserProperties(user));
       }
-    } else if (!_.get(user, "traits_customerio/deleted_at")) {
-      promises.push(
-        syncAgent.deleteUser(user)
-      );
+    } else if (!_.get(user, "traits_customerio/deleted_at") && !!_.get(user, "traits_customerio/id")) {
```
Comment (michaloo) [inline server/actions/update-user.js]: @mkawalek why you are doing that `!!` thing?

File: server/lib/sync-agent.js [hunk 2]
```diff
@@ -61,24 +61,35 @@ export default class SyncAgent {
     const created_at = Date.now() / 1000;
     const userIdent = { email };
 
-    const filteredAttributes = _.pick(user, this.userAttributesMapping);
+    const filteredSegments = _.pick(user, this.userAttributesMapping.map(a => a.hull));
```
Comment (michaloo) [inline server/lib/sync-agent.js]: `filteredSegments`? Isn't that `const` carrying actual traits picked from user?

File: server/actions/update-user.js (hunk 1 above)
Comment (mkawalek) [inline server/actions/update-user.js]: relict of the past :) I've removed it :)

File: server/lib/sync-agent.js (hunk 2 above)
Comment (mkawalek) [inline server/lib/sync-agent.js]: yes, it should be `filteredTraits`

File: package.json [hunk 3]
```diff
@@ -28,7 +28,7 @@
     "test:lint": "eslint server",
     "test:flow": "flow check",
     "test:unit": "NODE_ENV=test mocha -R spec ./tests/unit",
-    "test:integration": "NODE_ENV=test mocha -R spec ./tests/integration",
+    "test:integration": "NODE_ENV=test mocha -R spec ./tests/integration/index",
```
Comment (michaloo) [inline package.json]: oh, I didn't notice that before. @mkawalek please switch it for a command like here:
https://github.com/hull-ships/hull-intercom/blob/develop/package.json#L14

which add babel-register and run tests from all files in directory, I find it more clear to use `mocha` more "directly"

**Pull request merged**